src/language-id-client.ts

```diff
@@ -0,0 +1,135 @@
+import { LanguageIdJobOptions } from './models/language-id/LanguageIdJobOptions';
+import { LanguageIdResult } from './models/language-id/LanguageIdResult';
+
+const LanguageIdContentType = 'application/vnd.rev.languageid.v1.0+json';
+
+const TWO_GIGABYTES = 2e9; // Number of Bytes in 2 Gigabytes
+
+/**
+ * Client which handles connection to the Rev AI language id API.
+ */
+export class LanguageIdClient extends BaseApiClient<LanguageIdJob, LanguageIdResult> {
+    /**
+     * @param accessToken Access token used to validate API requests
+     * @param version (optional) version of the API to be used
+     */
+     constructor(accessToken: string) {
+        super(accessToken, 'languageid', 'v1');
+    }
+
+    /**
+     * See https://docs.rev.ai/api/language-identification/reference/#operation/GetLanguageIdentificationJobById
+     * Get information about a specific language id job
+     * @param id Id of job whose details are to be retrieved
+     * @returns Job details
+     */
+    async getJobDetails(id: string): Promise<LanguageIdJob> {
+        return super._getJobDetails(id);
+    }
+
+    /**
+     * See https://docs.rev.ai/api/language-identification/reference/#operation/GetListOfLanguageIdentificationJobs
+     * Get a list of language id jobs submitted within the last 30 days in reverse chronological order
+     * @param options Job options for getting list of jobs
+     * @returns List of job details
+     */
+    async getListOfJobs(options?: GetListOfJobsOptions): Promise<LanguageIdJob[]> {
+        return super._getListOfJobs(options);
+    }
+
+    /**
+     * See https://docs.rev.ai/api/language-identification/reference/#operation/DeleteLanguageIdentificationJobById
+     * Delete a specific language id job.
+     * All data related to the job, such as input and result, will be permanently deleted.
+     * A job can only by deleted once it's completed.
+     * @param id Id of job to be deleted
+     */
+    async deleteJob(id: string): Promise<void> {
+        return super._deleteJob(id);
+    }
+
+    /**
+     * See https://docs.rev.ai/api/language-identification/reference/#operation/SubmitLanguageIdentificationJob
+     * Submits a language id job with options.
+     * @param options Options submitted with the job: see LanguageIdJobOptions object
+     * @returns Details of the submitted job
+     */
+     async submitJob(options: LanguageIdJobOptions = {}): Promise<LanguageIdJob> {
+        return super._submitJob(options);
+    }
+
+    /**
+     * See https://docs.rev.ai/api/language-identification/reference/#operation/SubmitLanguageIdentificationJob
+     * Submits a language id job with a media url as input.
+     * @param mediaUrl Media url to be submitted for language id.
+     * @param options Options submitted with the job: see LanguageIdJobOptions object
+     * @returns Details of the submitted job
+     */
+    async submitJobUrl(mediaUrl: string, options: LanguageIdJobOptions = {}): Promise<LanguageIdJob> {
```

Comment:
@aaron-wilson-rev same here, if media_url is deprecated should we be allowing this?

Reply:
I changed it to accept source_config

Reply:
I don't think we need the submitJobUrl method at all for this client, since no one will be using it already. So here I think we could force usage of the submitJob with the job options set. If they still want to use media url in the job options they could for now but it would remove the url only submission option.